src/ui/common/src/components/workflows/WorkflowHeader.tsx

```diff
@@ -1,40 +1,45 @@
-import {
-  faCalendar,
-  faEllipsis,
-  faMicrochip,
-} from '@fortawesome/free-solid-svg-icons';
+import { faEllipsis, faMicrochip } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { Alert, Collapse, Tooltip, Typography } from '@mui/material';
 import Box from '@mui/material/Box';
-import React, { useEffect, useLayoutEffect, useState } from 'react';
-import Markdown from 'react-markdown';
+import React, { useLayoutEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
-import { visitParents } from 'unist-util-visit-parents';
 
+import {
+  useDagGetQuery,
+  useDagResultGetQuery,
+  useNodesGetQuery,
+  useWorkflowGetQuery,
+} from '../../handlers/AqueductApi';
 import { RootState } from '../../stores/store';
-import style from '../../styles/markdown.module.css';
 import { theme } from '../../styles/theme/theme';
-import { getNextUpdateTime } from '../../utils/cron';
 import { EngineType } from '../../utils/engine';
-import ExecutionStatus, { LoadingStatusEnum } from '../../utils/shared';
-import {
-  getWorkflowEngineTypes,
-  WorkflowDag,
-  WorkflowUpdateTrigger,
-} from '../../utils/workflows';
+import ExecutionStatus from '../../utils/shared';
+import { reduceEngineTypes } from '../../utils/workflows';
 import ResourceItem from '../pages/workflows/components/ResourceItem';
-import VersionSelector from './version_selector';
+import WorkflowDescription from './WorkflowDescription';
+import WorkflowNextUpdateTime from './WorkflowNextUpdateTime';
 import { StatusIndicator } from './workflowStatus';
+import VersionSelector from './WorkflowVersionSelector';
 
 export const WorkflowPageContentId = 'workflow-page-main';
 
 type Props = {
-  workflowDag: WorkflowDag;
+  apiKey: string;
+  workflowId: string;
+  dagId?: string;
+  dagResultId?: string;
 };
 
 const ContainerWidthBreakpoint = 700;
 
-const WorkflowHeader: React.FC<Props> = ({ workflowDag }) => {
+const WorkflowHeader: React.FC<Props> = ({
+  apiKey,
+  workflowId,
+  dagId,
+  dagResultId,
+}) => {
+  // TODO: Refactor
```

Comment:
Is there a Linear task for this? May be useful to include the task number on this todo comment